src/components/Header/Header.tsx

```diff
@@ -0,0 +1,69 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+import { DispatchContext, StateContext } from '../../store/todoReducer';
+import { Todo } from '../../types/state';
+import { Action } from '../../types/actions';
+
+export const Header = () => {
+  const titleRef = useRef<HTMLInputElement>(null);
+  const [todoTitle, setTodoTitle] = useState('');
+  const dispatch = useContext(DispatchContext);
+  const { todos } = useContext(StateContext);
+
+  useEffect(() => {
+    if (titleRef.current) {
+      titleRef.current.focus();
+    }
+  }, [todos]);
+
+  //I want to use useMemo, bit it doesn't work properly
```

Comment:
You don't need to useMemo in this case. It's we'll be over-engineering in this case. But if you want to practice memorization you can make it in the future.